aws/client.go

```diff
@@ -81,6 +82,9 @@ func (t *ResilientTransport) tries(req *http.Request) (res *http.Response, err e
 		if !t.ShouldRetry(req, res, err) {
 			break
 		}
+		if try == (t.MaxTries-1) {
+			break
+		}
```

Comment:
As I mentioned in the pull request comment, this cycle is broken. It wipes out response body before breaking.